cocotb/share/lib/simulator/simulatormodule.cpp

```diff
@@ -897,6 +910,9 @@ static PyMethodDef gpi_sim_hdl_methods[] = {
     {"get_signal_val_real",
         (PyCFunction)get_signal_val_real, METH_NOARGS,
         "Get the value of a signal as a double precision float"},
+    {"set_signal_val_int",
+        (PyCFunction)set_signal_val_int, METH_VARARGS,
+        "Set the value of a signal using an int"},
```

Comment:
Is this needed any more? Can we rename the other one to `int32`?

Reply:
@eric-wieser Actually, I think we want to keep both functions around to do different range checks. For #268 we want to support both int32_t range and uint32_t range from set_signal_value_long. I was thinking about renaming set_signal_value_long to set_signal_value_int_vec to distinguish it better from set_signal_value_int, but I'm open to suggestions.

Reply:
Are you actually after `set_signal_val_uint32` and `set_signal_val_int32`?

Reply:
```set_signal_val_int32``` for setting integerObjects and ```set_signal_val_uint32_int32``` for setting bitvectors up to 32 bit.

Reply:
@eric-wieser Changed everything to use ```set_signal_val_int()``` and introduced a 'limits' parameter to set the desired limits of the range check. Updated PR #1744 to also us a 'limits' parameter.
Is there anything else that needs to be done in this PR?